src/tests/ftest/launch.py

```diff
@@ -61,6 +61,32 @@
 )
 
 
+def log(message, details=None):
+    """Log the specified message.
+
+    Args:
+        message (str): string to log
+        details (str, optional): optional information to include with the error message if a
+            exception occurs during logging. Defaults to None.
+    """
+    retry_limit = 3
+    retry = 0
+    while retry <= retry_limit:
+        try:
+            print(message)
+            break
+        except (IOError, BlockingIOError) as error:
+            retry += 1
+            error_text = "Detected {}".format(error)
+            retry_text = "retry {}/{}".format(retry, retry_limit)
+            if details:
+                error_text = " ".join([error_text, details])
+            if retry <= retry_limit:
+                retry_text = "retry limit ({}) exceeded".format(retry_limit)
+                time.sleep(5)
```

Comment:
Did you mean?
```suggestion
            if retry >= retry_limit:
                retry_text = "retry limit ({}) exceeded".format(retry_limit)
                time.sleep(5)
```

Reply:
Reworked this to re-call the method for retries.